allensdk/brain_observatory/behavior/project_apis/data_io/behavior_project_cloud_api.py

```diff
@@ -0,0 +1,290 @@
+        row = row.squeeze()
+        data_path = self.cache.download_data(
+                str(int(row[self.cache.file_id_column])))
+        return BehaviorOphysExperiment.from_nwb_path(str(data_path))
+
+    def _get_session_table(self) -> pd.DataFrame:
+        session_table_path = self.cache.download_metadata(
+                "ophys_session_table")
+        self._session_table = pd.read_csv(session_table_path)
+
+    def get_session_table(self) -> pd.DataFrame:
+        """Return a pd.Dataframe table summarizing ophys_sessions
+        and associated metadata.
+
+        Notes
+        -----
+        - Each entry in this table represents the metadata of an ophys_session.
+        Link to nwb-hosted files in the cache is had via the
+        'ophys_experiment_id' column (can be a list)
+        and experiment_table
+        """
+        return self._session_table
+
+    def _get_behavior_only_session_table(self):
+        session_table_path = self.cache.download_metadata(
+                "behavior_session_table")
+        self._behavior_only_session_table = pd.read_csv(session_table_path)
+
+    def get_behavior_only_session_table(self) -> pd.DataFrame:
+        """Return a pd.Dataframe table with both behavior-only
+        (BehaviorSession) and with-ophys (BehaviorOphysExperiment)
+        sessions as entries.
+
+        Notes
+        -----
+        - In the first case, provides a critical mapping of
+        behavior_session_id to file_id, which the cache uses to find the
+        nwb path in cache.
+        - In the second case, provides a critical mapping of
+        behavior_session_id to a list of ophys_experiment_id(s)
+        which can be used to find file_id mappings in experiment_table
+        see method get_behavior_session()
+        - the BehaviorProjectCache calls this method through a method called
+        get_behavior_session_table. The name of this method is a legacy shared
+        with the behavior_project_lims_api and should be made consistent with
+        the BehaviorProjectCache calling method.
+        """
+        return self._behavior_only_session_table
+
+    def _get_experiment_table(self):
+        experiment_table_path = self.cache.download_metadata(
+                "ophys_experiment_table")
+        self._experiment_table = pd.read_csv(experiment_table_path)
+
+    def get_experiment_table(self):
+        """returns a pd.DataFrame where each entry has a 1-to-1
+        relation with an ophys experiment (i.e. imaging plane)
+
+        Notes
+        -----
+        - the file_id column allows the underlying cache to link
+        this table to a cache-hosted NWB file. There is a 1-to-1
+        relation between nwb files and ophy experiments. See method
+        get_behavior_ophys_experiment()
+        """
+        return self._experiment_table
+
+    def get_natural_movie_template(self, number: int) -> Iterable[bytes]:
+        """Download a template for the natural scene stimulus. This is the
```

Comment:
This docstring looks like it should be swapped with the one for `get_natural_scene_template()`?

Reply:
was a copy/paste.
fixed it everywhere.